src/CaptureClient/include/CaptureClient/CaptureEventProcessor.h

```diff
@@ -32,6 +32,10 @@ class CaptureEventProcessor {
 
   static std::unique_ptr<CaptureEventProcessor> CreateForCaptureListener(
       CaptureListener* capture_listener, absl::flat_hash_set<uint64_t> frame_track_function_ids);
+  static ErrorMessageOr<std::unique_ptr<CaptureEventProcessor>> CreateSaveToFileProcessor(
```

Comment:
Just to clarify, long term this one will replace the other one, right?

Reply:
Not in the foreseeable future, for now we want the UI client to do both save to file and process. There is going to be a combination/chain of CaptureEventProcessors